### DISCUSSION TITLE
Unexpected unknown

### DISCUSSION THREAD

**Discussion by davidcok** [cvc5/cvc5#7378] [Q&A] [answered]
CVC4 and CVC5 respond unknown instead of sat to this script:

(set-logic ALL)
(declare-fun p () Bool)
(declare-fun q () Bool)
(declare-fun r () Bool)
(declare-fun f (Bool Bool) Bool)
(assert (forall ((x Bool)(y Bool)(z Bool)) (=> (and (f x y) (f y z)) (f x z))))
(assert (and (f p q) (f q r)))
(assert (f p r))
(check-sat)
(exit)

While not 'wrong', it seems rather overly conservative. Z3 says sat.

**Command line arguments**: 
**cvc5 version/commit**: 0.0.2  
**Operating system**: MacOS

**Comment by ajreynol** [comment, marked as answer]
Hi David,
cvc5 will not try to answer "sat" for inputs with quantified formulas by default.  You can use `--finite-model-find` to enable strategies that will allow cvc5 to answer "sat" for benchmarks like this one.
Cheers,
Andy
